docs/sections/user_guide/cli/mode_config.rst

```diff
@@ -103,7 +175,253 @@ TBD
+    recipient: World
+    message: '{{ (greeting + " " + recipient) * repeat }}'
+    date: '{{ yyyymmdd }}'
+    repeat: 1
+    empty: 
+
+and supplemental YAML file ``config.yaml`` with content
+
+.. code:: sh
+
+  values:
+    greeting: Good Night
+    recipient: Moon
+    date: 
+    repeat: 2
+
+* Show the values in the input config file that have unfilled Jinja2 templates or empty keys:
+
+  .. code:: sh
+
+    $ uw config realize --input-file values.yaml --output-format yaml --values-needed
+    [2024-01-05T11:34:22]     INFO Keys that are complete:
+    [2024-01-05T11:34:22]     INFO     values
+    [2024-01-05T11:34:22]     INFO     values.greeting
+    [2024-01-05T11:34:22]     INFO     values.recipient
+    [2024-01-05T11:34:22]     INFO     values.message
+    [2024-01-05T11:34:22]     INFO     values.repeat
+    [2024-01-05T11:34:22]     INFO 
+    [2024-01-05T11:34:22]     INFO Keys that have unfilled Jinja2 templates:
+    [2024-01-05T11:34:22]     INFO     values.date: {{ yyyymmdd }}
+    [2024-01-05T11:34:22]     INFO 
+    [2024-01-05T11:34:22]     INFO Keys that are set to empty:
+    [2024-01-05T11:34:22]     INFO     values.empty
+
+* To realize the config to ``stdout``, a target output format must be explicitly specified:
+
+  .. code:: sh
+
+    $ uw config realize --input-file values.yaml --output-format yaml
+    values:
+      greeting: Hello
+      recipient: World
+      message: Hello World
+      date: '{{ yyyymmdd }}'
+      repeat: 1
+      empty: null
+
+  Shell redirection via ``|``, ``>``, et al may also be used to stream output to a file, another process, etc.
+
+* Values in the primary input file can be overridden via one or more supplemental files specified as positional arguments, each overriding the last; or by environment variables, which have the highest precedence.
+
+  .. code:: sh
+
+    $ recipient=Sun uw config realize --input-file values.yaml --output-format yaml config.yaml 
+    values:
+      greeting: Good Night
+      recipient: Moon
+      message: Good Night SunGood Night Sun
+      date: 20240105
+      repeat: 2
+      empty: null
+
+* Realize the config to a file via command-line argument:
+
+  .. code:: sh
+
+    $ uw config realize --input-file values.yaml --output-file realized.yaml config.yaml
+
+  The content of ``realized.yaml``:
+
+  .. code:: sh
+
+    values:
+      greeting: Good Night
+      recipient: Moon
+      message: Good Night MoonGood Night Moon
+      date: 20240105
+      repeat: 2
+      empty: null
+
+* With the ``--dry-run`` flag specified, nothing is written to ``stdout`` (or to a file if ``--output-file`` is specified), but a report of what would have been written is logged to ``stderr``:
+
+  .. code:: sh
+
+    $ uw config realize --input-file values.yaml --output-file realized.yaml --dry-run config.yaml
+    [2024-01-05T11:35:20]     INFO values:
+    values:
+      greeting: Good Night
+      recipient: Moon
+      message: Good Night MoonGood Night Moon
```

Comment:
nitpick: I think the run-on text without a space where one would expect one between the two copies of the greeting makes this seem like a somewhat tortured example. Like, who would want that? I just tried and wasn't readily able to get a space in there -- I tried piping into `| join(" ")` but that just put a space between each character -- probably a list of two strings is needed here but I don't know how to get one and Jinja2 isn't making it easy.

I **do** like that this examples shows the use of Jinja2 beyond the simplest variable interpolation, and don't want to use that, or else I'd suggest removing the `repeat`. But since this `Good Night MoonGood Night Moon` is now repeated so many times on this page, I feel like we should try for a strong Jinja2 example that doesn't produce output that looks like a bug. I don't immediately have a suggestion, unfortunately.

Reply:
I could make the `message` formula `'{{ (greeting + " " + recipient + " ") * repeat` just to add an extra space